tools/cmake/preset/default.cmake

```diff
@@ -249,7 +249,7 @@ check_required_options_on(
 
 check_conflicting_options_on(
   IF_ON EXECUTORCH_BUILD_ARM_BAREMETAL CONFLICTS_WITH
-  EXECUTORCH_BUILD_EXTENSION_DATA_LOADER EXECUTORCH_BUILD_PTHREADPOOL
```

Comment:
I suspect this or `install_executorch` used by the ci scripts was the reason it was working before? 

If we want to not enable this by default then perhaps we should also update `install_executorch` (if not using the default preset already). 

And that should also make the baremetal build fail. I also realized that we don't run the baremetal size-test binary on the CI, so we may need to fix that too.

Reply:
> the reason it was working before?

you may be interested in later PRs in this stack, particularly #12744